src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -866,6 +867,9 @@ internal async Task SearchPackagesAndRefreshUpdateCountAsync(string searchText,
                     pSearchCallback.ReportComplete(searchTask, (uint)searchResult.PackageSearchItems.Count);
                 }
 
+                sw.Stop();
+                EmitRefreshEvent(TimeSpan.Zero, RefreshOperationSource.PackageManagerLoaded, RefreshOperationStatus.NoOp, isUIFiltering: false, sw.Elapsed.TotalMilliseconds);
```

Comment:
I don't think this is the right place. 

The purpose of the refresh event is to capture the time at which is it got triggered and capture the different entry points that trigger the refresh. 

Now we're emitting in the refresh itself, so now you're likely getting 2 events per, which isn't right afaik.

Reply:
> I don't think this is the right place.
> 
> The purpose of the refresh event is to capture the time at which is it got triggered and capture the different entry points that trigger the refresh.
> 
> Now we're emitting in the refresh itself, so now you're likely getting 2 events per, which isn't right afaik.

2 events is fine. 1st one will be emitted without duration which original behavior, so we don't care that one. 2nd one is emitted with duration which we care about. Also please note I'm not resetting timespace with `GetTimeSinceLastRefreshAndRestart` on 2nd  one.

Reply:
Another solution would be I just create new event instead of piggybacking on this EmitRefreshEvent.

Reply:
>2 events is fine

I don't think it is. As indicated by my explanation of what the event does. 

Don't be afraid to make a change in 5-6 spots, if it for the right reasons, such as preserving context of the refresh. 

> Another solution would be I just create new event instead of piggybacking on this EmitRefreshEvent.

I think the refresh event is the most appropriate one here. I think it's good to build on existing telemetry.

Reply:
Already commented in other thread, but my approach was wrapping the loaded event here: https://github.com/NuGet/NuGet.Client/commit/e837672effa93a0faee968ba324632f9d439bf6c#diff-80991ce1edef4f242608751b70dd2d48dad8e24a28dbefd2b68a3a7784a12f21R408

Reply:
@nkolev92 
Please check latest update. But numbers coming with this one doesn't feel accurate what I see on screen. Problem is below code adding quite a lot of duration even though PMUI is already populated.

https://github.com/NuGet/NuGet.Client/blob/7a5b0aafd767d105014c995378a1dbb71a5d86bd/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#L870-L873

How about I add my previous measurement with new event type besides this one?
So that way we get best of both worlds, refresh event will capture duration current tab populate and other tabs update also current tab only too.

Reply:
> Please check latest update. But numbers coming with this one doesn't feel accurate what I see on screen (I'm testing with my phone stopwatch). Problem is below code adding quite a lot of duration even though PMUI is already populated.

What's the difference between the measurements here and before?
They look equivalent. 

> Problem is below code adding quite a lot of duration even though PMUI is already populated.

Is that indicative of this current implementation, or indicative of the way we're trying to measure things. 

> How about I add my previous measurement with new event type besides this one?
So that way we get best of both worlds, refresh event will capture duration current tab populate and other tabs update also current tab only too.

As I mentioned above, not sure how they're different. 
My feedbawck was about the reporting the event, not how it's being measured.

Reply:
> > Please check latest update. But numbers coming with this one doesn't feel accurate what I see on screen (I'm testing with my phone stopwatch). Problem is below code adding quite a lot of duration even though PMUI is already populated.
> 
> What's the difference between the measurements here and before? They look equivalent.

It's about sometimes 30 sec vs 62 on Consolidate Tab, not the all-time maybe something to do with caching. I feel 30 sec is more accurate, but remaining time spent on call HttpCalls. Below is Task view while waiting for it

![image](https://user-images.githubusercontent.com/8766776/155627928-703a4800-5dca-49c4-8035-a51e5e67fe59.png)

> 
> > Problem is below code adding quite a lot of duration even though PMUI is already populated.
> 
> Is that indicative of this current implementation, or indicative of the way we're trying to measure things.

I think happening due to current implementation.

> 
> > How about I add my previous measurement with new event type besides this one?
> > So that way we get best of both worlds, refresh event will capture duration current tab populate and other tabs update also current tab only too.
> 
> As I mentioned above, not sure how they're different. My feedbawck was about the reporting the event, not how it's being measured.

`RefreshInstalledAndUpdatesTabsAsync` is making ton of http calls and holding things about another 30 sec in my case.

Reply:
So you're saying the difference is in https://github.com/NuGet/NuGet.Client/blob/7a5b0aafd767d105014c995378a1dbb71a5d86bd/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#L870-L873 right? 

Do those calls not affect what's shown on UI? 
If they do affect what's shown UI, then they should be captured right? 
Wouldn't the current implementation be more correct then?

Reply:
> So you're saying the difference is in
> 
> https://github.com/NuGet/NuGet.Client/blob/7a5b0aafd767d105014c995378a1dbb71a5d86bd/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#L870-L873
> 
> right?

Yes, right.

> Do those calls not affect what's shown on UI? If they do affect what's shown UI, then they should be captured right? Wouldn't the current implementation be more correct then?

Now I agree. Could you be able to review current implementation?